packages/kbn-esql-editor/src/esql_editor.tsx

```diff
@@ -444,28 +443,31 @@ export const ESQLEditor = memo(function ESQLEditor({
     };
   }, [esqlCallbacks, code]);
 
-  const clientParserStatus = clientParserMessages.errors?.length
-    ? 'error'
-    : clientParserMessages.warnings.length
-    ? 'warning'
-    : 'success';
+  const clientParserStatus = getStatusFromMessages(clientParserMessages);
 
   useEffect(() => {
     const validateQuery = async () => {
       if (editor1?.current) {
         const parserMessages = await parseMessages();
-        setClientParserMessages({
+        const nextClientParserMessages = {
           errors: parserMessages?.errors ?? [],
           warnings: parserMessages?.warnings ?? [],
+        };
+        const nextClientParserStatus = getStatusFromMessages(nextClientParserMessages);
+        addQueriesToCache({
+          queryString: code,
+          status: nextClientParserStatus,
```

Comment:
@stratoula It seems that there is a race condition in setting a status for the new history item. 

Wdyt about this approach?

I don't quite understand the dependency of this useEffect on `clientParserStatus` and what other use cases which might affect it. Also  when `editor1?.current` could be unavailable?

It might be better if code owners take a close look at how this race condition can be avoided .

Reply:
@stratoula Alright, thanks!